src/lib/annotations/doc/DocDrawingThread.js

```diff
@@ -1,78 +1,88 @@
 import {
+    STATES,
     STATES_DRAW,
     PAGE_PADDING_TOP,
     PAGE_PADDING_BOTTOM,
     CLASS_ANNOTATION_LAYER_DRAW
 } from '../annotationConstants';
-import { getScale } from '../annotatorUtil';
 import DrawingPath from '../drawing/DrawingPath';
 import DrawingThread from '../drawing/DrawingThread';
 import * as docAnnotatorUtil from './docAnnotatorUtil';
+import * as annotatorUtil from '../annotatorUtil';
 
-class DocDrawingThread extends DrawingThread {
-    /** @property {number} - Drawing state */
-    lastPage;
+const CLASS_ANNOTATION_LAYER_DRAW_MEMORY = `${CLASS_ANNOTATION_LAYER_DRAW}-mem`;
 
+class DocDrawingThread extends DrawingThread {
     /** @property {HTMLElement} - Page element being observed */
     pageEl;
 
     //--------------------------------------------------------------------------
     // Public
     //--------------------------------------------------------------------------
     /**
+     * [constructor]
+     *
+     * @inheritdoc
+     * @param {AnnotationThreadData} data - Data for constructing thread
+     * @return {DocDrawingThread} Drawing annotation thread instance
+     */
+    constructor(data) {
+        super(data);
+
+        this.handleStart = this.handleStart.bind(this);
+        this.handleMove = this.handleMove.bind(this);
+        this.handleStop = this.handleStop.bind(this);
+        this.checkAndHandleScaleUpdate = this.checkAndHandleScaleUpdate.bind(this);
+        this.reconstructBrowserCoordFromLocation = this.reconstructBrowserCoordFromLocation.bind(this);
```

Comment:
I'll have @tonyjin and @pramodsum weigh in on this. I prefer this, but it does seem cumbersome